packages/smooth_app/lib/pages/product/product_page/header/reorder_bottom_sheet.dart

```diff
@@ -0,0 +1,181 @@
+import 'package:flutter/material.dart';
+import 'package:provider/provider.dart';
+import 'package:smooth_app/generic_lib/bottom_sheets/smooth_bottom_sheet.dart';
+import 'package:smooth_app/generic_lib/design_constants.dart';
+import 'package:smooth_app/resources/app_icons.dart' as icons;
+import 'package:smooth_app/themes/smooth_theme.dart';
+import 'package:smooth_app/themes/smooth_theme_colors.dart';
+
+typedef LabelBuilder<T> = Widget Function(
+  BuildContext context,
+  T item,
+  int index,
+);
+
+class ReorderBottomSheet<T> extends StatelessWidget {
+  ReorderBottomSheet({
+    required List<T> items,
+    required this.onReorder,
+    required this.labelBuilder,
+    this.onVisibilityToggle,
+    required this.title,
+  }) : _items = items.map((T data) => _ReorderableItem<T>(data: data)).toList();
+
+  final List<_ReorderableItem<T>> _items;
+  final ValueChanged<List<T>> onReorder;
+  final LabelBuilder<T> labelBuilder;
+  final ValueChanged<T>? onVisibilityToggle;
+  final String title;
+
+  @override
+  Widget build(BuildContext context) {
+    final SmoothColorsThemeExtension theme =
+        context.extension<SmoothColorsThemeExtension>();
+
+    return ChangeNotifierProvider<_ReorderBottomSheetProvider<T>>(
+      create: (_) => _ReorderBottomSheetProvider<T>(_items),
+      child: Consumer<_ReorderBottomSheetProvider<T>>(
+        builder:
+            (BuildContext context, _ReorderBottomSheetProvider<T> provider, _) {
+          return DraggableScrollableSheet(
+            expand: false,
+            initialChildSize: 0.6,
+            maxChildSize: 0.9,
+            minChildSize: 0.4,
+            builder: (_, ScrollController scrollController) {
+              return ClipRRect(
+                borderRadius: const BorderRadius.vertical(top: ROUNDED_RADIUS),
+                child: Container(
+                  decoration: BoxDecoration(
+                    color: Theme.of(context).canvasColor,
+                    borderRadius: const BorderRadius.vertical(
+                      top: Radius.circular(16),
+                    ),
+                  ),
+                  child: Column(
+                    crossAxisAlignment: CrossAxisAlignment.start,
+                    children: <Widget>[
+                      const SmoothModalSheetHeader(
+                        title: 'Reorder tabs',
+                      ),
+                      Expanded(
+                        child: ReorderableListView.builder(
+                          scrollController: scrollController,
+                          padding: const EdgeInsets.all(MEDIUM_SPACE),
```

Comment:
EdgeInsetsDirectional

Reply:
This doesn't seem to be possible as `ReorderableListView` takes an explicit `EdgeInsets` parameter and not an `EdgeInsetsGeometry`

Reply:
No pb, you can ignore my comment